python/MoltresXS.py

```diff
@@ -0,0 +1,280 @@
+                    for j in range(len(val)):
+                        temps = np.append(temps, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+
+            if 'Nominal' in line:
+                val = lines[k+2].split()
+                k += 2
+                temps = np.append(temps, float(val[0]))
+            if 'Burnups' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        burn = np.append(burn, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+            if 'Betas' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        beta = np.append(beta, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+                if not 'num_delay_groups' in locals():
+                    num_delay_groups = len(beta)
+            if 'Lambdas' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        decay = np.append(decay, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+            if 'total-transfer' in line:
+                val = lines[k+1].split()
+                remxs = np.append(remxs, float(val[1]))
+
+                val = lines[k+3].split()
+                fissxs = np.append(fissxs, float(val[0]))
+                nsf = np.append(nsf, float(val[3]))
+                fissE = np.append(nsf, float(val[4]))
+
+                val = lines[k+5].split()
+                difcoef = np.append(difcoef, float(val[2]))
+                chi = np.append(chi, float(val[1]))
+
+                val = lines[k+7].split()
+                invV = np.append(invV, float(val[4]))
+                k += 7
+            if 'Scattering cross' in line:
+                while 1:
+                    val = lines[k+1].split()
+                    k += 1
+                    for j in range(len(val)):
+                        transxs = np.append(transxs, float(val[j]))
+                    if "'" in lines[k+1]:
+                        break
+            k += 1
+        num_temps = len(temps)
+        if num_temps == 0:
+            num_temps = 1
+            temps = 600*np.ones(1)
+        self.xs_lib = {}
+        for i in range(num_burn):
+            self.xs_lib[i] = {}
+            for j in range(num_uni):
+                self.xs_lib[i][j] = {}
+                for k in range(num_temps):
+                    self.xs_lib[i][j][k] = {}
+                    index = i*(num_temps*num_uni*num_groups)+k * \
+                        (num_groups*num_uni)+j*num_groups
+                    self.xs_lib[i][j][k]["REMXS"] = remxs[index:index+num_groups]
```

Comment:
The repetition here in the next 20 lines indicates a place where code-reuse would allow you to loop through many similar functions rather than copy paste.